src/heuristic/HeuristicMapper.cpp

```diff
@@ -302,19 +301,40 @@ void HeuristicMapper::mapUnmappedGates(long layer, HeuristicMapper::Node& node,
 			}
 
 			if (possibleEdges.empty()) {
-				throw QMAPException("Could not map logical qubits to physical qubits. No suitable edge found.");
+                double bestScore = std::numeric_limits<int>::max();
+                std::pair<unsigned short, unsigned short> bestMapping;
+                for(int i=0; i < architecture.getNqubits(); i++) {
+                    for (int j = i + 1; j < architecture.getNqubits(); j++) {
+                        if (qubits.at(i) == DEFAULT_POSITION && qubits.at(j) == DEFAULT_POSITION) {
+                            double dist = architecture.distance(i, j);
+                            if (dist < bestScore) {
+                                bestScore = dist;
+                                bestMapping = std::make_pair(i, j);
+                            }
+                        }
+                    }
+                }
```

Comment:
Could this possibly be a member function of the `Architecture` class?
More generally: would it make sense to add functions to the architecture class that return the shortest/longest path?
Of course this is out of scope for this PR, so feel free to ignore it for now.

Reply:
Yeah, this could be moved to the Architecture class. However, as we currently only use this at this place, I'd do this as refactor later as this is too specific for general usage at the moment.